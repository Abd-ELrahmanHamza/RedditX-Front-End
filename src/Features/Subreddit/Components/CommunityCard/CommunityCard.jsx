// import { ReactComponent as CakeIcon } from "Features/Subreddit/Assets/piece-of-cake-2-svgrepo-com.svg";
import AboutCommunity from "../AboutCommunity/AboutCommunity";
import CommunityDate from "../CommunityDate/CommunityDate";
import CommunityOptions from "../CommunityOptions/CommunityOptions";
import CommunityTopics from "../CommunityTopics/CommunityTopics";
import OnlineMember from "../OnlineMembers/OnlineMembers";
import UserFlairPreview from "../UserFlairPreview/UserFlairPreview";
import { Container, Content, Separator } from "./CommunityCard.styled";

const CommunityCard = () => {
  return (
    <Container>
      <AboutCommunity />
      <Content>
        <CommunityDate />
        <Separator />
        <OnlineMember />
        {true && <CommunityTopics />}
        <Separator />
        <UserFlairPreview />
        <CommunityOptions />
      </Content>
    </Container>
  );
};

export default CommunityCard;
