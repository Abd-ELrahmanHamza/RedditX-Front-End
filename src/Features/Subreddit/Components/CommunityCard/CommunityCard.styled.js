import styled from "styled-components";

export const Container = styled.div`
  background-color: #fff;
  color: #1a1a1b;
  fill: #1a1a1b;
  border: 1px solid #ccc;
  border-radius: 4px;
  overflow: visible;
  word-wrap: break-word;
`;

export const Content = styled.div`
  max-height: none;
  padding: 12px;
`;

export const Separator = styled.hr`
  background-color: rgba(26, 26, 27, 0.07);
  border: none;
  height: 1px;
  margin: 16px 0;
  color: #1a1a1c;
  opacity: 1;
`;
