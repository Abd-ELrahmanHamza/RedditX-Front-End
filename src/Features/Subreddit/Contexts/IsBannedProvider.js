import { createContext, useContext, useState } from "react";
const isBannedContext = createContext(null);

export const IsBannedProvider = ({ children }) => {
  const [isBanned, setIsBanned] = useState(null);

  return (
    <isBannedContext.Provider value={{ isBanned, setIsBanned }}>
      {children}
    </isBannedContext.Provider>
  );
};

export const useIsBanned = () => {
  return useContext(isBannedContext);
};
