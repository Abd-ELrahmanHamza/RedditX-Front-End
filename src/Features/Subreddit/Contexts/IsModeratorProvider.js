import { createContext, useContext, useState } from "react";
const isModeratorContext = createContext(null);

export const IsModeratorProvider = ({ children }) => {
  const [isMod, setIsMod] = useState(null);

  return (
    <isModeratorContext.Provider value={{ isMod, setIsMod }}>
      {children}
    </isModeratorContext.Provider>
  );
};

export const useIsModerator = () => {
  return useContext(isModeratorContext);
};
