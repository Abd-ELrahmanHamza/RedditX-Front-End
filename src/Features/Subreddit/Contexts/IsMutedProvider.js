import { createContext, useContext, useState } from "react";
const isMutedContext = createContext(null);

export const IsMutedProvider = ({ children }) => {
  const [isMuted, setIsMuted] = useState(null);

  return (
    <isMutedContext.Provider value={{ isMuted, setIsMuted }}>
      {children}
    </isMutedContext.Provider>
  );
};

export const useIsMuted = () => {
  return useContext(isMutedContext);
};
