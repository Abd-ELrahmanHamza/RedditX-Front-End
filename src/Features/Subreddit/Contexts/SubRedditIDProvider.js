import { createContext, useContext, useState } from "react";
const subRedditIDContext = createContext(null);

export const SubRedditIDProvider = ({children }) => {

  const [communityID, setCommunityID] = useState(null);

  return (
    <subRedditIDContext.Provider value={{ communityID, setCommunityID}}>
      {children}
    </subRedditIDContext.Provider>
  );
};

export const useSubRedditID = () => {
  return useContext(subRedditIDContext);
};