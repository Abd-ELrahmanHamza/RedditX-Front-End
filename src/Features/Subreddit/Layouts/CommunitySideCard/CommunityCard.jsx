import {
BtnDiv, 
CommunityContainer, 
ViewAllBtn,
CoverDiv,
GradientDiv,
LinkH2,
} from "./CommunityCard.styled";
import { useNavigate } from "react-router-dom";
import { CommunityOl } from "../Communities Container/CommunitiesContainer.styled";
import CommunitySideCardItem from "../../Components/CommunitySideCardItem/CommunityCardItem";
import axios from "API/axios";
import useFetch from "Hooks/useFetch";
import {Link} from "react-router-dom";
/**
 * Component that links each  of community card item.
 *
 * @Component
 * @returns {React.Component}
 */
const CommunitySideCard = ({communityList}) => {

    const navigate = useNavigate();
    
    const navigateToRandomCat = (categoryTitle) => {
        // 👇️ navigate to /contacts
        navigate(`/category/${categoryTitle}`);
    };
    let cat = 'Growing';    //Assign Random Category  [Can change to be a prop to change shown random among the two cards]
    let com; 

    if(communityList && communityList.length!==0) {
        
        cat = communityList.communities[0]['category'];   //Assign the Random Category to be shown [can call two requests in leaderboard Page, to have different radnoms in both cards]
        com = communityList.communities.slice(0,5).map((community, index) => {
            return (
                <li key={community._id}>
                    <CommunitySideCardItem
                    title = {community.description}
                    img = {community.icon}
                    index = {index+1}/>
                </li>
            );
        });
    }
    // const cat =
    
    return (
        <CommunityContainer>
              <CoverDiv>
                <GradientDiv>
                  <LinkH2>
                      <Link href={"https://www.reddit.com/subreddits/leaderboard/"}>Top
                          <span>&nbsp;{cat}&nbsp;</span>
                          Communities
                      </Link>
                  </LinkH2>
                </GradientDiv>
              </CoverDiv>
              <CommunityOl>
                  {com}
              </CommunityOl>
            <BtnDiv>
                <ViewAllBtn onClick={()=>navigateToRandomCat(cat)}>See All {cat}</ViewAllBtn>
            </BtnDiv>
        </CommunityContainer>
    );
};

export default CommunitySideCard;