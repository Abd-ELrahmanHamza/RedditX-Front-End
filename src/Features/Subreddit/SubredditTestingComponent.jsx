import { ThemeProvider } from "styled-components";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";
import { BrowserRouter as Router } from "react-router-dom";
import { SubRedditProvider } from "Features/Subreddit/Contexts/SubRedditProvider";
import { SubRedditIDProvider } from "./Contexts/SubRedditIDProvider";
import { IsModeratorProvider } from "Features/Subreddit/Contexts/IsModeratorProvider";
import { IsBannedProvider } from "Features/Subreddit/Contexts/IsBannedProvider";
import { IsMutedProvider } from "Features/Subreddit/Contexts/IsMutedProvider";

const theme = { ...defaultTheme, ...lightTheme };

/**
 * Function that wraps Children props inside a Browser Router and a Theme Provider
 * @returns {React.Component}The component is being returned.
 */
const SubredditTestingComponent = ({ children }) => {
  return (
    <Router>
      <ThemeProvider theme={theme}>
        <SubRedditProvider>
          <SubRedditIDProvider>
            <IsModeratorProvider>
              <IsBannedProvider>
                <IsMutedProvider>{children}</IsMutedProvider>
              </IsBannedProvider>
            </IsModeratorProvider>
          </SubRedditIDProvider>
        </SubRedditProvider>
      </ThemeProvider>
    </Router>
  );
};

export default SubredditTestingComponent;
