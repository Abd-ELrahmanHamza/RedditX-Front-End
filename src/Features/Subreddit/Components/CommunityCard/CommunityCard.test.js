import CommunityCard from "./CommunityCard";
import { render } from "@testing-library/react";
import { useSubReddit } from "Features/Subreddit/Contexts/SubRedditProvider";
import SubredditTestingComponent from "Features/Subreddit/SubredditTestingComponent";
import { useEffect } from "react";

const MockCard = () => {
  const { setCommunity } = useSubReddit();
  useEffect(() => {
    setCommunity([
      {
        communityId: "t1_54148d012f459b337e2382ce",
        name: "Community",
        memberCount: 552,
        rank: 0,
        createdAt: "2022/02/15, 15:02:20",
        icon: "photo.jpg",
        banner: "photo.jpg",
        communityRules: [
          {
            title: "rule1",
            textDescription: "This rule1!",
          },
          {
            title: "rule2",
            textDescription: "This rule2!",
          },
          {
            title: "rule3",
            textDescription: "",
          },
          {
            title: "rule4",
            textDescription: "This rule4!",
          },
        ],
        FAQs: {
          question: "string",
          answer: "string",
        },
        communityOptions: {
          privacyType: "string",
          region: "string",
          postType: "string",
          allowImgLinkUploads: true,
          suggestedCommentSort: "string",
          allowMultipleImagePerPost: true,
          welcomeMessage: true,
          NSFW: true,
          emailUserNameMention: true,
          optionsDescription: "string",
        },
        communityDescription: "this is a Subreddit description",
      },
    ]);
  }, []);
  return <CommunityCard />;
};

describe("Content of Subreddit", () => {
  it("should render Content of Subreddit without crashing", () => {
    render(
      <SubredditTestingComponent>
        <MockCard />
      </SubredditTestingComponent>
    );
  });
});
