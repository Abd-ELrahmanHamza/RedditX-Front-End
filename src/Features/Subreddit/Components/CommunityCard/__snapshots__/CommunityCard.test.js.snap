// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the card of community should render without crashing 1`] = `ShallowWrapper {}`;
